examples/interpolate_custom_schedule.rs

```diff
@@ -0,0 +1,419 @@
+    };
+
+    // Add the `TransformInterpolationPlugin` to the app to enable transform interpolation.
+    app.add_plugins((
+        DefaultPlugins,
+        BackgroundFixedUpdatePlugin::<task_user::TaskWorkerTraitImpl>::default(),
+        easing_plugin,
+        interpolation_plugin,
+    ));
+
+    // Setup the scene and UI, and update text in `Update`.
+    app.add_systems(Startup, (setup, setup_text)).add_systems(
+        bevy::app::prelude::RunFixedMainLoop,
+        (
+            change_timestep,
+            update_timestep_text,
+            update_diff_to_render_text,
+        ),
+    );
+
+    app.add_systems(
+        bevy::app::prelude::RunFixedMainLoop,
+        (ease_translation_lerp, ease_rotation_slerp, ease_scale_lerp)
+            .in_set(TransformEasingSet::Ease),
+    );
+
+    // Run the app.
+    app.run();
+}
+
+/// Eases the translations of entities with linear interpolation.
+fn ease_translation_lerp(
+    mut query: Query<(&mut Transform, &TranslationEasingState)>,
+    time: Query<(&TaskToRenderTime, &Timestep)>,
+) {
+    let Ok((time, timestep)) = time.get_single() else {
+        return;
+    };
+    let overstep = (time.diff.max(0.0) / timestep.timestep.as_secs_f64()).min(1.0) as f32;
+    query.iter_mut().for_each(|(mut transform, interpolation)| {
+        if let (Some(start), Some(end)) = (interpolation.start, interpolation.end) {
+            transform.translation = start.lerp(end, overstep);
+        }
+    });
+}
+
+/// Eases the rotations of entities with spherical linear interpolation.
+fn ease_rotation_slerp(
+    mut query: Query<(&mut Transform, &RotationEasingState)>,
+    time: Query<(&TaskToRenderTime, &Timestep)>,
+) {
+    let Ok((time, timestep)) = time.get_single() else {
+        return;
+    };
+    let overstep = (time.diff.max(0.0) / timestep.timestep.as_secs_f64()).min(1.0) as f32;
+
+    query
+        .par_iter_mut()
+        .for_each(|(mut transform, interpolation)| {
+            if let (Some(start), Some(end)) = (interpolation.start, interpolation.end) {
+                // Note: `slerp` will always take the shortest path, but when the two rotations are more than
+                // 180 degrees apart, this can cause visual artifacts as the rotation "flips" to the other side.
+                transform.rotation = start.slerp(end, overstep);
+            }
+        });
+}
+
+/// Eases the scales of entities with linear interpolation.
+fn ease_scale_lerp(
+    mut query: Query<(&mut Transform, &ScaleEasingState)>,
+    time: Query<(&TaskToRenderTime, &Timestep)>,
+) {
+    let Ok((time, timestep)) = time.get_single() else {
+        return;
+    };
+    let overstep = (time.diff.max(0.0) / timestep.timestep.as_secs_f64()).min(1.0) as f32;
+
+    query.iter_mut().for_each(|(mut transform, interpolation)| {
+        if let (Some(start), Some(end)) = (interpolation.start, interpolation.end) {
+            transform.scale = start.lerp(end, overstep);
+        }
+    });
+}
```

Comment:
Those systems could use a helper function to only retrieve the `overstep()`, the rest of the code is similar to the easing plugin.